build_tools/ci/use_xrt_directly/run_test.sh

```diff
@@ -0,0 +1,193 @@
+else
+  MLIR_AIE_INSTALL=`realpath "$3"`
+fi
+if [ ! -d "${MLIR_AIE_INSTALL}" ]; then
+  echo "No directory '${MLIR_AIE_INSTALL}' (argument 3) found."
+  exit 1
+fi
+
+# Parameter 4) <peano-install-dir>
+if [ -z "${4-}" ]; then
+  PEANO=/opt/llvm-aie
+else
+  PEANO=`realpath "$4"`
+fi
+if [ ! -d "${PEANO}" ]; then
+  echo "No directory '${PEANO}' (argument 4) found."
+  exit 1
+fi
+
+# Parameter 5) <xrt-dir>
+if [ -z "${5-}" ]; then
+  XRT_DIR=/opt/xilinx/xrt
+else
+  XRT_DIR=`realpath "$5"`
+fi
+if [ ! -d "${XRT_DIR}" ]; then
+  echo "No directory '${XRT_DIR}' (argument 5) found."
+  exit 1
+fi
+
+# Parameter 6) <vitis-install-dir>
+if [ -z "${6-}" ]; then
+  # An alternate to a full vitis install, will work
+  # here but not for a full build of mlir-aie
+  # https://riallto.ai/install-riallto.html#install-riallto
+  # VITIS=/opt/Riallto/Vitis/2023.1
+  VITIS=/opt/Xilinx/Vitis/2023.2
+else
+  VITIS=`realpath "$6"`
+fi
+if [ ! -d "${VITIS}" ]; then
+  echo "No directory '${VITIS}' (argument 6) found."
+  exit 1
+fi
+
+THIS_DIR="$(cd $(dirname $0) && pwd)"
+ROOT_DIR="$(cd $THIS_DIR/../.. && pwd)"
+
+source $XRT_DIR/setup.sh
+
+THIS="$(cd $(dirname $0) && pwd)"
+SOURCE_MLIR_FILE="${THIS_DIR}/linalg_matmul.mlir"
+
+# Construct the iree-compile command. 
+IREE_COMPILE_COMMAND="${IREE_COMPILE_EXE} \
+${SOURCE_MLIR_FILE} \
+--iree-hal-target-backends=amd-aie \
+--iree-amd-aie-peano-install-dir=${PEANO} \
+--iree-amd-aie-mlir-aie-install-dir=${MLIR_AIE_INSTALL} \
+--iree-amd-aie-vitis-install-dir=${VITIS} \
+--iree-hal-dump-executable-files-to=${OUTPUT_DIR} \
+--iree-amdaie-use-pipeline=simple-pack \
+-o ${OUTPUT_DIR}/test_artefact.vmfb \
+--iree-amd-aie-show-invoked-commands"
+
+# Execute the command to generate the .vmfb, .xclbin, .ipu.txt files, etc.
+echo "Executing command: $IREE_COMPILE_COMMAND"
+eval $IREE_COMPILE_COMMAND 
+
+if [ ! -f "${OUTPUT_DIR}/test_artefact.vmfb" ]; then
+  echo "test_artefact.vmfb was not created: ${OUTPUT_DIR}/test_artefact.vmfb"
+  exit 1
+fi
+
+
+# Get the full path of test.cpp (which is in the same directory as this script).
+TEST_CPP="${THIS_DIR}/test.cpp"
+
+
+# Compile the host code (flags inspired by mlir-air tests). 
+g++ ${TEST_CPP} -o ${OUTPUT_DIR}/test.exe -Wall -I${XRT_DIR}/include -L${XRT_DIR}/lib -luuid -lxrt_coreutil -lrt -lstdc++
+
+# Verify that ${OUTPUT_DIR}/test.exe exists:
+if [ ! -f "${OUTPUT_DIR}/test.exe" ]; then
+  echo "test.exe was not created: ${OUTPUT_DIR}/test.exe"
+  exit 1
+fi
+
+# We expect to find a file with the .xclbin extension in the output directory, 
+# or in a subdirectory of the output directory. Locate it if possible. 
+XCLBIN_FILE=""
+for file in `find ${OUTPUT_DIR} -name "*.xclbin"`; do
+  XCLBIN_FILE="${file}"
+done
+if [ -z "${XCLBIN_FILE}" ]; then
+  echo "No .xclbin file found in the output directory or any of its subdirectories."
+  exit 1
+else
+  echo "Found .xclbin file: ${XCLBIN_FILE}"
+fi
+
+# Maybe sign the XCLBIN file (CI moonshot). 
```

Comment:
@erieaton-amd here I sign the XCLBIN directly, without first copying it and then signing the copy. 

Question about the clean-up: how does copying it with a new name ensure all xclbins are removed in CI? Based on a comment on discord.  Is what I'm doing here bad? How I can see what is actually happening on the CI machine, how can I check what junk is being created and left there? I guess that turned into 3 questions... 

FYI @daveliddell

Reply:
This is cluttering the CI machine. The renaming scheme is expected by the cleanup phase to make sure that we delete every kernel that got copied to the firmware directory by this github job and not any other github jobs.

Reply:
I understand now. 

Repeating what you already know to help me remember... $SIGNER moves the xclbin it "signs" to a directory outside of github's CI temporary directory. It doesn't sign in-place. The place it moves to is always the same, so it's a problem if multiple CI jobs run in parallel -- the signer is moving xcblins from different locations (each CI starts in a different temporary directory) to the same location -- race conditions. 

To prevent this, each xclbin is given a unique name, based on it's job id. This uniquely named xclbin is what is then copied to the "special place" in /lib/firmware for signing. 

So that resolves the race condition, but not the problem of cleaning up the garbage left in /lib/firmware. This is done in another CI script.